### ISSUE TITLE
[java] [regression] false positive UnnecessaryImport (not happening with 6.55.0)

### ISSUE THREAD

**Issue by kohlschuetter** [pmd/pmd#4613]
**Affects PMD Version:** 7.0.0-SNAPSHOT

**Rule:** [UnnecessaryImport](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**
PMD 7 doesn't detect the use of these imports.

**Code Sample demonstrating the issue:**

```java
AcceptTimeoutTest.java
```

**Expected outcome:**
no errors/warnings

PMD reports violations, but that's wrong. That's a false positive.:

```
AcceptTimeoutTest.java:20:	UnnecessaryImport:	Unused static import 'org.junit.jupiter.api.Assertions.assertNotNull'
AcceptTimeoutTest.java:21:	UnnecessaryImport:	Unused static import 'org.junit.jupiter.api.Assertions.assertThrows'
AcceptTimeoutTest.java:22:	UnnecessaryImport:	Unused static import 'org.junit.jupiter.api.Assertions.assertTimeoutPreemptively'
AcceptTimeoutTest.java:23:	UnnecessaryImport:	Unused static import 'org.junit.jupiter.api.Assertions.assertTrue'
AcceptTimeoutTest.java:24:	UnnecessaryImport:	Unused static import 'org.junit.jupiter.api.Assertions.fail'
AcceptTimeoutTest.java:26:	UnnecessaryImport:	Unused import 'java.io.IOException'
AcceptTimeoutTest.java:28:	UnnecessaryImport:	Unused import 'java.net.Socket'
AcceptTimeoutTest.java:30:	UnnecessaryImport:	Unused import 'java.net.SocketException'
AcceptTimeoutTest.java:31:	UnnecessaryImport:	Unused import 'java.net.SocketTimeoutException'
AcceptTimeoutTest.java:32:	UnnecessaryImport:	Unused import 'java.time.Duration'
```

**Running PMD through:** CLI, Maven

CLI:
```
pmd check AcceptTimeoutTest.java -f text -R pmd-ruleset.xml      
```

Also occurs with Maven when using the compatibility patch for the Maven plugin from PR https://github.com/pmd/pmd/pull/4609

Java file and ruleset attached.
[file-and-ruleset.zip](https://github.com/pmd/pmd/files/11909184/file-and-ruleset.zip)

**Comment by adangel** [comment]
This is due to missing auxclasspath. If AcceptTimeoutTest is compiled (and all referenced classes are available), then I get no false positives from the rule UnnecessaryImport.

The auxclasspath needs to be provided on the CLI via `--aux-classpath`. When running through maven/gradle, this should be provided automatically (e.g. all dependencies).
